tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionReplayTransportFactory.cs

```diff
@@ -0,0 +1,121 @@
+// <copyright file="ExceptionReplayTransportFactory.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Debugger;
+using Datadog.Trace.HttpOverStreams;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.ExceptionAutoInstrumentation
+{
+    internal static class ExceptionReplayTransportFactory
+    {
+        private const string DefaultRelativePath = "/api/v2/debugger";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ExceptionReplayTransportFactory));
+
+        internal static ExceptionReplayTransportInfo Create(TracerSettings tracerSettings, ExceptionReplaySettings settings, IDiscoveryService discoveryService)
+        {
+            if (!settings.AgentlessEnabled)
+            {
+                return CreateAgentTransport(tracerSettings, discoveryService);
+            }
+
+            if (string.IsNullOrWhiteSpace(settings.AgentlessApiKey))
+            {
+                Log.Error("Exception Replay agentless uploads enabled but DD_API_KEY is not set. Disabling Exception Replay.");
+                throw new InvalidOperationException("Exception Replay agentless mode requires DD_API_KEY.");
```

Comment:
Do we really need/want to throw here, or should we just return null and bail out in the callee without throwing? We've already logged the error, so I don't think it's necessary?